hawkular-inventory-api/src/main/java/org/hawkular/inventory/api/Datas.java

```diff
@@ -0,0 +1,145 @@
+import org.hawkular.inventory.api.paging.Order;
+import org.hawkular.inventory.api.paging.Page;
+import org.hawkular.inventory.api.paging.Pager;
+
+/**
+ * @author Lukas Krejci
+ * @since 0.3.0
+ */
+public final class Datas {
+
+    private Datas() {
+
+    }
+
+    protected interface ReadBase {
+
+        /**
+         * @return the access to the resource this data belongs to.
+         */
+        Resources.Single resource();
+    }
+
+    public interface Read extends ReadInterface<Single, Multiple, Void>, ReadBase {
+    }
+
+    public interface ReadWrite
+            extends ReadWriteInterface<DataEntity.Update, DataEntity.Blueprint, Single, Multiple, Void>,
+            ReadBase {
+    }
+
+    public interface Single extends ResolvableToSingle<DataEntity, DataEntity.Update> {
+
+        /**
+         * Loads the data entity on the current position in the inventory traversal along with its data.
+         *
+         * <p>Note that this might be a potentially expensive operation because of the attached data structure being
+         * loaded.
+         *
+         * @return the fully loaded structured data on the current position in the inventory traversal
+         * @throws EntityNotFoundException if there is no structured data on the current position in the inventory
+         * @see ResolvableToSingle#entity()
+         */
+        @Override
+        DataEntity entity() throws EntityNotFoundException;
+
+        /**
+         * Returns the data on the path relative to the entity.
+         *
+         * In another words, you can use this method to obtain only a subset of the data stored on the data entity.
+         *
+         * <p>I.e if you have an data entity which contains a map with a key "foo", which contains a list and you
+         * want to obtain a third element of that list, you'd do:
+         * {@code
+         * ...data(RelativePath.to().structuredData().key("foo").index(2).get());
+         * }
+         *
+         * <p>If you want to obtain the whole data structure, use an empty path: {@code RelativePath.empty().get()}.
+         *
+         * @param dataPath the path to the subset of the data.
+         * @return the subset of the data stored with the data entity
+         * @see #bareData(RelativePath)
+         */
+        StructuredData data(RelativePath dataPath);
+
+        /**
+         * This is very similar to {@link #data(RelativePath)} but this method doesn't load the child data.
+         *
+         * <p>If the data on the path contains a "primitive" value, the value is loaded. If the data contains a list,
+         * the returned instance will contain an empty list and if the data contains a map the returned instance will
+         * contain an empty map.
+         *
+         * @param dataPath the path to the subset of the data to return
+         * @return the subset of the data stored with the data entity
+         */
+        StructuredData bareData(RelativePath dataPath);
```

Comment:
This is here because of performance reasons, right? Perhaps the `flatData()` would be more descriptive name here.